src/game/client/components/scoreboard.cpp

```diff
@@ -342,80 +469,131 @@ void CScoreboard::OnRender()
+			RenderGoals(Width/2-w-1.5f, y+3.0f+ScoreboardHeight, w*2.0f+3.0f);
+
+			// scoreboard size
+			m_TotalRect.x = Width/2-w-1.5f;
+			m_TotalRect.y = y;
+			m_TotalRect.w = w*2.0f+3.0f;
+			m_TotalRect.h = ScoreboardHeight+SpectatorHeight+3.0f;
+		}
+	}
 
-				float w = TextRender()->TextWidth(0, 86.0f, aText, -1);
-				TextRender()->Text(0, Width/2-w/2, 39, 86.0f, aText, -1);
+	Width = 400*3.0f*Graphics()->ScreenAspect();
+	Graphics()->MapScreen(0, 0, Width, 400*3.0f);
+	if(m_pClient->m_Snap.m_pGameData && (m_pClient->m_GameInfo.m_GameFlags&GAMEFLAG_TEAMS) && m_pClient->m_Snap.m_pGameDataTeam)
+	{
+		const char *pRedClanName = GetClanName(TEAM_RED);
+		const char *pBlueClanName = GetClanName(TEAM_BLUE);
+
+		if(m_pClient->m_Snap.m_pGameData->m_GameStateFlags&GAMESTATEFLAG_GAMEOVER)
+		{
+			char aText[256];
+			str_copy(aText, Localize("Draw!"), sizeof(aText));
+
+			if(m_pClient->m_Snap.m_pGameDataTeam->m_TeamscoreRed > m_pClient->m_Snap.m_pGameDataTeam->m_TeamscoreBlue)
+			{
+				if(pRedClanName)
+					str_format(aText, sizeof(aText), Localize("%s wins!"), pRedClanName);
+				else
+					str_copy(aText, Localize("Red team wins!"), sizeof(aText));
 			}
-			else if(m_pClient->m_Snap.m_pGameData->m_GameStateFlags&GAMESTATEFLAG_ROUNDOVER)
+			else if(m_pClient->m_Snap.m_pGameDataTeam->m_TeamscoreBlue > m_pClient->m_Snap.m_pGameDataTeam->m_TeamscoreRed)
 			{
-				char aText[256];
-				str_copy(aText, Localize("Round over!"), sizeof(aText));
-
-				float w = TextRender()->TextWidth(0, 86.0f, aText, -1);
-				TextRender()->Text(0, Width/2-w/2, 39, 86.0f, aText, -1);
+				if(pBlueClanName)
+					str_format(aText, sizeof(aText), Localize("%s wins!"), pBlueClanName);
+				else
+					str_copy(aText, Localize("Blue team wins!"), sizeof(aText));
 			}
 
-			RenderScoreboard(Width/2-w-5.0f, 150.0f, w, TEAM_RED, pRedClanName ? pRedClanName : Localize("Red team"));
-			RenderScoreboard(Width/2+5.0f, 150.0f, w, TEAM_BLUE, pBlueClanName ? pBlueClanName : Localize("Blue team"));
+			float tw = TextRender()->TextWidth(0, FontSize, aText, -1);
+			TextRender()->Text(0, Width/2-tw/2, 39, FontSize, aText, -1);
 		}
+		else if(m_pClient->m_Snap.m_pGameData->m_GameStateFlags&GAMESTATEFLAG_ROUNDOVER)
+		{
+			char aText[256];
+			str_copy(aText, Localize("Round over!"), sizeof(aText));
 
-		RenderGoals(Width/2-w/2, 150+760+10, w);
-		RenderSpectators(Width/2-w/2, 150+760+10+50+10, w);
+			float tw = TextRender()->TextWidth(0, FontSize, aText, -1);
+			TextRender()->Text(0, Width/2-tw/2, 39, FontSize, aText, -1);
+		}
 	}
 
 	RenderRecordingNotification((Width/7)*4);
 }
 
+void CScoreboard::OnMessage(int MsgType, void *pRawMsg)
+{
+	if(MsgType == NETMSGTYPE_SV_KILLMSG)
+	{
+		CNetMsg_Sv_KillMsg *pMsg = (CNetMsg_Sv_KillMsg *)pRawMsg;
+
+		m_aPlayerStats[pMsg->m_Victim].m_Deaths++;
+		if(pMsg->m_Victim != pMsg->m_Killer)
+			m_aPlayerStats[pMsg->m_Killer].m_Kills++;
+	}
+	else if(MsgType == NETMSGTYPE_SV_CLIENTDROP && Client()->State() != IClient::STATE_DEMOPLAYBACK)
```

Comment:
should be:
`else if(MsgId == NETMSGTYPE_SV_CLIENTINFO && Client()->State() != IClient::STATE_DEMOPLAYBACK)
	{
		CNetMsg_Sv_ClientInfo *pMsg = (CNetMsg_Sv_ClientInfo *)pRawMsg;

		if(!m_pClient->m_aClients[pMsg->m_ClientID].m_Active)
		{
				return;
		}
		ResetPlayerStats(pMsg->m_ClientID);
	}`

Reply:
done